### PR TITLE
Fix/30042 fix lhn previews and report actions v2

### PR DESCRIPTION
**Pull request by koko57**
### Fixed Issues

$ https://github.com/Expensify/App/issues/30042
PROPOSAL: https://github.com/Expensify/App/issues/30042#issuecomment-1792625104

### Tests

**Create an iouReport**

1. Global Create > Request Money > $50 > choose a person > Confirm
2. Land in the DM
3. Look at the LHN row to observe the preview

request action preview (requestor view)

>Tom Rhys Jones owes $50.00
>requested 10.00

request action preview (payer view)

>Tom Rhys Jones owes $50.00
>Agata: requested 10.00

DMs:
owed iouReport being previewed (Agata view - DM)

>Tom Rhys Jones
>Tom Rhys Jones owes $50.00

owed iouReport being previewed (Tom view - DM)

>Agata Kosior
>Tom Rhys Jones owes $50.00

**Create an expenseReport**

1. Create a workspace if you need to
2. Global Create > Request Money > $50 > choose the workspace > confirm
3. Land in the workspace chat
5. Look at the LHN row to observe the preview

Request action preview (admin viewer)

>Duraflame owes $100.00
>Tom: requested $50.00

Request action preview (requestor viewer)

>Duraflame owes $100.00
>requested $50.00

owed expenseReport being previewed (member view - workspace chat)

>Duraflame
>Duraflame owes $50.00

owed expenseReport being previewed (admin view - workspace chat)

>Tom Rhys Jones
>Duraflame owes $50.00

**Pay an iouReport**

1. Navigate to the iouReport view
2. Click Pay elsewhere green button at the top right of the report

paid action preview (requestor view)

>Tom Rhys Jones paid $50.00
>Tom: paid $50.00 elsewhere

paid action preview (payer view)

>Tom Rhys Jones paid $50.00
>paid $50.00 elsewhere

paid iouReport being previewed (Agata view - DM)

>Tom Rhys Jones
>Tom Rhys Jones paid $50.00

paid iouReport being previewed (Tom view - DM)

>Agata Kosior
>Tom Rhys Jones paid $50.00

**Pay an expenseReport**

1. Navigate to the expenseReport view
2. Click Pay elsewhere green button at the top right of the report
5. Observe the system message added to the expenseReport
6. Observe the LHN preview

Paid action preview (admin viewer)

>Duraflame paid $100.00
>paid $100.00 elsewhere

paid action preview (requestor viewer)

>Duraflame paid $100.00
>Agata: paid $100.00 elsewhere

paid expenseReport being previewed (member view - workspace chat)

>Duraflame
>Duraflame paid $50.00

paid expenseReport being previewed (admin view - workspace chat)

>Tom Rhys Jones
>Duraflame paid $50.00

**Add a comment** 

comment preview (sender view)

>Tom Rhys Jones owes $50.00 <--- note: this is obviously variable dependent on if the report title is owed or paid
>blah, blah, I love apples!

comment preview (receiver view)

>Tom Rhys Jones owes $50.00 <--- note: this is obviously variable dependent on if the report title is owed or paid
>Tom: blah, blah, I love apples!

comment preview (comment sender viewer)

>Duraflame owes $100.00 <--- note: this is obviously variable dependent on if the report title is owed or paid
>blah blah, I love bananas!

comment preview (comment receiver viewer)

>Duraflame owes $100.00 <--- note: this is obviously variable dependent on if the report title is owed or paid
>Tom: blah blah, I love bananas!

- Verify that other previews are not affected
- Verify that the amount changes properly (i.e. when requesting in other currency)

1. Go to 1:1 DM.
2. Click + > Request money.
3. Create a manual request using a different currency.
4. Click on the IOU preview to open IOU report.
5. Note that the amount for the IOU report in LHN shows the local currency instead of the requested currency.

or when changing the amount

1. Go to 1:1 DM.
2. Click + > Request money.
3. Create a manual request.
4. Click on the IOU preview twice to open IOU details page.
5. Click Amount and change the amount.
6. Save the new amount.

- Verify that app doesn't crash on requesting money, going offline/online
- Verify that the preview is displaying correctly for the workspace member

1. Create a workspace in admin and invite a member
2. In member account , Global Create > Request Money > $50 > choose the workspace > confirm
3. As a admin navigate to the expenseReport view
4. Click Pay elsewhere green button at the top right of the report
5. Observe the system message added to the expenseReport
6. Login to member account and observe the LHN preview

- Verify that the app doesn't crash when requesting from workspace second time

1. Request 1:1 money as employee from the admin
2. Do second request 1:1 money as employee from the admin

- [x] Verify that no errors appear in the JS console

### Offline tests

1. Create a request offline and verify the it's displayed correctly
2. Pay for a request offline and verify the it's displayed correctly

### QA Steps

Same as in Test Steps

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

Screenshot_1703140718

</details>

<details>
<summary>Android: mWeb Chrome</summary>

Screenshot_1703140842

</details>

<details>
<summary>iOS: Native</summary>

Simulator Screen Shot - iPhone 14 Pro - 2023-12-20 at 18 53 15

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

I don't know why but I'm having problem with opening the app on iOS Safari

 **EXPENSE REPORT**
 
 **Request action preview** (admin viewer)
 
> Duraflame owes $100.00
> Tom: requested $50.00

Screenshot 2023-12-20 at 17 47 28
 
**Request action preview** (requestor viewer)
 
> Duraflame owes $100.00
> requested $50.00

Screenshot 2023-12-20 at 17 47 04
 
**Paid action preview** (admin viewer)
 
> Duraflame paid $100.00
> paid $100.00 elsewhere

Screenshot 2023-12-20 at 18 21 25

**paid action preview** (requestor viewer)
 
> Duraflame paid $100.00
> Agata: paid $100.00 elsewhere
 
Screenshot 2023-12-20 at 18 18 40

**comment preview** (comment sender viewer)

> Duraflame owes $100.00   _<--- note: this is obviously variable dependent on if the report title is `owed` or `paid`_
> blah blah, I love bananas!

Screenshot 2023-12-20 at 17 52 17

**comment preview** (comment receiver viewer)

> Duraflame owes $100.00   _<--- note: this is obviously variable dependent on if the report title is `owed` or `paid`_
> Tom: blah blah, I love bananas!

Screenshot 2023-12-20 at 17 52 32
 

**IOU REPORT** So that same logic will apply to iouReports as well. For brevity, the below examples are just Agata looking at a variety of actions Tom took only to highlight the `:` in use.

**request action preview**

> Tom Rhys Jones owes $50.00
> Tom: requested 10.00

Screenshot 2023-12-20 at 18 00 38

**paid action preview**
 
> Tom Rhys Jones paid $50.00
> Tom: paid $50.00 elsewhere

Screenshot 2023-12-20 at 18 36 47

**comment preview**
 
> Tom Rhys Jones owes $50.00  _<--- note: this is obviously variable dependent on if the report title is `owed` or `paid`_
> Tom: blah, blah, I love apples!

receiver

Screenshot 2023-12-20 at 18 03 42

sender

Screenshot 2023-12-20 at 18 03 32

 
**DM / WORKSPACE CHAT**
 
As for the DM/Workspace Chat LHN preview message when the expense/iouReport is the lastMessage being previewed, those don't use a colon separator because none of the chat participants are the "author" of the report preview component being previewed. So those just use the report title of the expense/iouReport being previewed.

**owed iouReport being previewed** (Agata view - DM)
 
> Tom Rhys Jones
> Tom Rhys Jones owes $50.00

Screenshot 2023-12-20 at 18 00 23

**owed iouReport being previewed** (Tom view - DM)

> Agata Kosior
> Tom Rhys Jones owes $50.00

Screenshot 2023-12-20 at 18 00 49
 
**paid iouReport being previewed** (Agata view - DM)

> Tom Rhys Jones
> Tom Rhys Jones paid $50.00

Screenshot 2023-12-20 at 18 02 29
 
**paid iouReport being previewed** (Tom view - DM)
 
> Agata Kosior
> Tom Rhys Jones paid $50.00

Screenshot 2023-12-20 at 18 02 10
 
**owed expenseReport being previewed** (member view - workspace chat)

> Duraflame
> Duraflame owes $50.00

Screenshot 2023-12-20 at 17 46 58

**owed expenseReport being previewed** (admin view - workspace chat)

> Tom Rhys Jones
> Duraflame owes $50.00
 
Screenshot 2023-12-20 at 17 57 19

**paid expenseReport being previewed** (member view - workspace chat)

> Duraflame
> Duraflame paid $50.00

Screenshot 2023-12-20 at 17 50 29

**paid expenseReport being previewed** (admin view - workspace chat)

> Tom Rhys Jones
> Duraflame paid $50.00

Screenshot 2023-12-20 at 17 49 58

<details>
<summary>MacOS: Desktop</summary>

Screenshot 2023-12-20 at 19 08 39

</details>

### CODE REVIEW

File: src/libs/SidebarUtils.ts [hunk 1]
```diff
@@ -331,7 +333,11 @@ function getOptionData(
               }
             : null;
     }
-    const lastActorDisplayName = hasMultipleParticipants && lastActorDetails?.accountID && Number(lastActorDetails.accountID) !== currentUserAccountID ? lastActorDetails.displayName : '';
+
+    const lastActorDisplayName =
```
Comment (adhorodyski) [inline src/libs/SidebarUtils.ts:337]: Having these inline makes it more and more hard to follow - have you considered abstracting these checks away to a few functions? You can even colocate them nearby if they're not really reusable, but having eg. `isCurrentActorLast()` etc will be less cognitively demanding. Not a blocker though, this lgtm :)

File: src/libs/SidebarUtils.ts [hunk 2]
```diff
@@ -354,7 +360,9 @@ function getOptionData(
         }
     }
 
-    if ((result.isChatRoom || result.isPolicyExpenseChat || result.isThread || result.isTaskReport) && !result.isArchivedRoom) {
+    const isThreadMessage = ReportUtils.isThread(report) && reportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.ADDCOMMENT;
+
+    if ((result.isChatRoom || result.isPolicyExpenseChat || result.isThread || result.isTaskReport || isThreadMessage) && !result.isArchivedRoom) {
```
Comment (adhorodyski) [inline src/libs/SidebarUtils.ts:365]: this raises pretty much the same concerns

Comment (koko57) [reply src/libs/SidebarUtils.ts:366]: This one I'll leave for another refactor

File: src/libs/ReportUtils.ts [hunk 3]
```diff
@@ -2031,6 +2038,18 @@ function getReportPreviewMessage(
...
 
     const containsNonReimbursable = hasNonReimbursableTransactions(report.reportID);
+
+    const lastActorID = reportAction?.actorAccountID;
+
+    // if we have the amount in the originalMessage and lastActorID, we can use that to display the preview message for the latest request
+    if (originalMessage?.amount !== undefined && lastActorID && !isPreviewMessageForParentChatReport) {
+        const amount = originalMessage?.amount;
+        const currency = originalMessage?.currency ?? report.currency ?? '';
+        const amountToDisplay = CurrencyUtils.convertToDisplayString(Math.abs(amount), currency);
```
Comment (mountiny) [inline src/libs/ReportUtils.ts:2048]: In theory the expense report total or request could be negative value, so we should not use Abs here but more so invert the value if expense.

However, I think its not a concern for now

File: src/libs/ReportUtils.ts [hunk 4]
```diff
@@ -2031,6 +2038,18 @@ function getReportPreviewMessage(
...
+        const requestorName = lastActorID && lastActorID !== currentUserAccountID ? getDisplayNameForParticipant(lastActorID, !isPreviewMessageForParentChatReport) : '';
+        return `${requestorName ? `${requestorName}: ` : ''}${Localize.translateLocal('iou.requestedAmount', {formattedAmount: amountToDisplay})}`;
```
Comment (mountiny) [inline src/libs/ReportUtils.ts:2050]:
```suggestion

        // We only want to show the actor name in the preview if its not the current user who took the action
        const requestorName = lastActorID && lastActorID !== currentUserAccountID ? getDisplayNameForParticipant(lastActorID, !isPreviewMessageForParentChatReport) : '';
        return `${requestorName ? `${requestorName}: ` : ''}${Localize.translateLocal('iou.requestedAmount', {formattedAmount: amountToDisplay})}`;
```

**Review by mountiny — COMMENTED** [review]
looking good, one comment

**Comment by mananjadhav** [comment]
Starting review now

File: src/libs/ReportUtils.ts (hunk 3 above)
Comment (koko57) [reply src/libs/ReportUtils.ts:2048]: @mountiny so can I leave it for now as it is?

**Comment by mountiny** [comment]
@koko57 One change requested, there is an extra space in the IOU/ expense report in front of the `paid` message because there used to be name 
image

You can see it in the video above too

**Comment by koko57** [comment]
@mountiny as even in this [snippet](https://github.com/Expensify/App/blob/d6f359fc010fe8120cf8a6761a4dac21c819e270/src/libs/ReportUtils.ts#L4072) the payer's name could be an empty string and we're using it for this 3 translation keys I will add condition to display them with or without additional space

**Review by mananjadhav — APPROVED** [review]
Took a bit of time but the code changes and Tests look fine. I'll need a day to finish the whole checklist, but approving due to urgency @mountiny

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

File: src/libs/ReportUtils.ts (hunk 3 above)
Comment (mountiny) [reply src/libs/ReportUtils.ts:2048]: Yes, just noting

**Review by mountiny — APPROVED** [review]
Thank you!

**Review by mountiny — APPROVED** [review]
These changes are platform agnostic and I have tested thoroughly on web and we should be ready to move on, I am merging now before the deploy so we can have Staging QA done on this overnight

Thank you @koko57 for sticking with this one
